sdcflows/utils/wrangler.py

```diff
@@ -531,6 +540,39 @@ def find_estimators(
                     _log_debug_estimation(logger, e, layout.root)
                     estimators.append(e)
 
+                medic_entities = {**base_entities, **{'part': 'mag', 'echo': Query.ANY}}
+                has_magnitude = tuple()
+                with suppress(ValueError):
+                    has_magnitude = layout.get(
+                        suffix='bold',
+                        **medic_entities,
+                    )
+
+                for mag_img in has_magnitude:
+                    phase_img = layout.get(**{**mag_img.get_entities(), **{'part': 'phase'}})
+                    if not phase_img:
+                        continue
+
+                    phase_img = phase_img[0]
+                    try:
+                        e = fm.FieldmapEstimation(
+                            [
+                                fm.FieldmapFile(mag_img.path, metadata=mag_img.get_metadata()),
+                                fm.FieldmapFile(phase_img.path, metadata=phase_img.get_metadata()),
+                            ]
+                        )
+                    except (ValueError, TypeError) as err:
+                        _log_debug_estimator_fail(
+                            logger,
+                            "potential MEDIC fieldmap",
+                            [mag_img, phase_img],
```

Comment:
Then here we want to collect all of the images, right?

```suggestion
                    complex_imgs = layout.get(**{**mag_img.get_entities(), **{'part': ['phase', 'mag']}})

                    if complex_imgs[0].path in fm._estimators.sources:
                        continue

                    try:
                        e = fm.FieldmapEstimation(
                            [
                                fm.FieldmapFile(img.path, metadata=img.get_metadata()) for img in complex_imgs
                            ]
                        )
                    except (ValueError, TypeError) as err:
                        _log_debug_estimator_fail(
                            logger,
                            "potential MEDIC fieldmap",
                            complex_imgs,
```